tests/test_wps_resolve_rules.py

```diff
@@ -7,32 +7,24 @@
 
 @pytest.mark.online
 @pytest.mark.parametrize(
-    ("csv", "date_range", "region", "geoserver", "ensemble", "thredds",),
+    ("csv", "date_range", "region", "geoserver", "ensemble",),
     [
         (
             resource_filename("tests", "data/rules_small.csv"),
             "2050",
             "vancouver_island",
             "http://docker-dev01.pcic.uvic.ca:30123/geoserver/bc_regions/ows",
             "p2a_rules",
-            "False",
-        ),
-        (
-            resource_filename("tests", "data/rules_small.csv"),
-            "2050",
-            "vancouver_island",
-            "http://docker-dev01.pcic.uvic.ca:30123/geoserver/bc_regions/ows",
-            "p2a_rules",
-            "True",
         ),
     ],
 )
+@pytest.mark.parametrize("thredds", [True, False])
 def test_wps_resolve_rules(
     mock_thredds_url_root, csv, date_range, region, geoserver, ensemble, thredds,
 ):
     with open(csv, "r") as csv_file:
         datainputs = (
-            f"csv_content={csv_file.read()};"
+            f"csv={csv_file.read()};"
```

Comment:
You still have to use `read()`?

Reply:
Yes. Using `.read()` gives us `FileIO` and the raw url gives us `StringIO`, both of which can be handled by `stream`.